esp-hal/src/gpio/mod.rs

```diff
@@ -286,53 +216,54 @@ pub trait InputPin: Pin {
         signal: InputSignal,
         invert: bool,
         force_via_gpio_mux: bool,
-    ) -> &mut Self;
+        _: private::Internal,
+    );
 
     /// Remove a connected `signal` from this input pin.
     ///
     /// Clears the entry in the GPIO matrix / Io mux that associates this input
     /// pin with the given [input `signal`](`InputSignal`). Any other
     /// connected signals remain intact.
-    fn disconnect_input_from_peripheral(&mut self, signal: InputSignal) -> &mut Self;
+    fn disconnect_input_from_peripheral(&mut self, signal: InputSignal, _: private::Internal);
 }
 
 /// Trait implemented by pins which can be used as outputs
 pub trait OutputPin: Pin {
     /// Configure open-drain mode
-    fn set_to_open_drain_output(&mut self) -> &mut Self;
+    fn set_to_open_drain_output(&mut self, _: private::Internal);
```

Comment:
returning `&mut Self` makes it inconvenient for delegation - since it's an internal API I don't think this is a problem